core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -279,9 +279,10 @@ private static boolean isCreate(UpdateTableRequest request) {
   }
 
   private static TableMetadata create(TableOperations ops, TableMetadata start, UpdateTableRequest request) {
-    TableMetadata.Builder builder = TableMetadata.buildFrom(start);
-
     // the only valid requirement is that the table will be created
+    request.requirements().forEach(requirement -> requirement.validate(ops.current()));
```

Comment:
I added this so the assertion for validation check exception message is consistent with other validation checks.